bin/agd

```diff
@@ -196,9 +196,10 @@ case "${unameOut}" in
     Darwin*)    platform=mac;;
     *)          platform=win;;
 esac
+
 # check the xsnap version against our baked-in notion of what version we should be using
 xsnap_version=$("${thisdir}/../packages/xsnap/xsnap-native/xsnap/build/bin/${platform}/release/xsnap-worker" -n)
-[[ "${xsnap_version}" == "agoric-upgrade-10" ]] || fatal "xsnap out of date"
+[[ "${xsnap_version}" == "${XSNAP_VERSION}" ]] || fatal "xsnap version mismatch; expected ${XSNAP_VERSION}, got ${xsnap_version}"
```

Comment:
I suggest moving this entire version check up above the:

```sh
if $BUILD_ONLY; then
  echo "Build complete." 1>&2
  exit 0
fi
```

condition.  That way, it will be invoked even if you run `agd build`, which is likely a better time to fail rather than just when trying to run other agd subcommands.